rgw/v2/tests/aws/test_aws.py

```diff
@@ -70,6 +70,48 @@ def test_exec(config, ssh_con):
             if config.test_ops.get("enable_version", False):
                 log.info(f"bucket versioning test on bucket: {bucket_name}")
                 aws_reusable.put_get_bucket_versioning(bucket_name, endpoint)
+        if config.test_ops.get("verify_etag_for_complete_multipart_upload", False):
+            log.info(
+                f"Verifying ETag element for complete multipart upload is not empty string"
+            )
+            log.info("Create multipart upload")
+            key_name = bucket_name + "_object"
+            create_mp_upload_resp = aws_reusable.create_multipart_upload(
+                bucket_name, key_name, endpoint
+            )
+            upload_id = json.loads(create_mp_upload_resp)["UploadId"]
+            log.info(f"upload part one of object: {key_name}")
+            utils.exec_shell_cmd(f"fallocate -l 20m key20m")
+            upload_part1 = json.loads(
+                aws_reusable.upload_part(
+                    bucket_name, key_name, 1, upload_id, "key20m", endpoint
+                )
+            )
+            log.info(f"upload part two of object: {key_name}")
+            utils.exec_shell_cmd(f"fallocate -l 22m key22m")
+            upload_part2 = json.loads(
+                aws_reusable.upload_part(
+                    bucket_name, key_name, 2, upload_id, "key22m", endpoint
+                )
+            )
+
+            parts = [
+                {"ETag": upload_part1["ETag"], "PartNumber": 1},
+                {"ETag": upload_part2["ETag"], "PartNumber": 2},
+            ]
+            mpstructure = {"Parts": parts}
+            os.system("touch test.json")
+            with open("test.json", "w") as fd:
+                json.dump(mpstructure, fd)
+            complete_multipart_upload_resp = json.loads(
+                aws_reusable.complete_multipart_upload(
+                    bucket_name, key_name, "test.json", upload_id, endpoint
+                )
+            )
+            if not complete_multipart_upload_resp["ETag"]:
+                raise AssertionError(
+                    "Etag not generated during complete multipart upload operation"
+                )
```

Comment:
@ckulal , could you place this code in a separate method "upload_multipart_object" for reusability and take the object size from config 
as present in https://github.com/red-hat-storage/ceph-qe-scripts/blob/master/rgw/v2/tests/s3_swift/reusable.py#L450

Reply:
Done.
New log: http://magna002.ceph.redhat.com/ceph-qe-logs/Chaithra/aws_etag_verification/aws_etag_verification.log-new